### PR TITLE
Ria 1956 personal details

### PR DESCRIPTION
**Pull request by chrisgrimble**
Allow the user to enter personal details. Updated validation.

### JIRA link (if applicable) ###

https://tools.hmcts.net/jira/browse/RIA-1956

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[X] No
```

### CODE REVIEW

File: app/controllers/personal-details.ts [hunk 1]
```diff
@@ -53,7 +58,9 @@ function postDateOfBirth(updateAppealService: UpdateAppealService) {
 
 function getNamePage(req: Request, res: Response, next: NextFunction) {
   try {
-    const { personalDetails } = req.session.appeal.application || null;
+    const personalDetails = req.session.appeal.application.personalDetails && (req.session.appeal.application.personalDetails.givenNames || req.session.appeal.application.personalDetails.familyName) ?
+      req.session.appeal.application.personalDetails : { givenNames: req.idam.userDetails.forename, familyName: req.idam.userDetails.surname };
+
```
Comment (t0nin0s) [inline app/controllers/personal-details.ts]: Can we please move this logic to loadAppeal in update-appeal-service?

File: app/utils/fields-validations.ts [hunk 2]
```diff
@@ -14,10 +14,11 @@ function validate(obj: object, schema: any): ValidationErrors | null {
...
   if (result.error) {
     return result.error.details.reduce((acc, curr): ValidationError => {
-      acc[curr.context.key] = {
-        key: curr.context.key,
+      const key = curr.context.key || (curr.context.peers ? curr.context.peers.join('-') : curr.context.peers) ;
+      acc[key] = {
+        key: key,
         text: curr.message,
-        href: `#${curr.context.key}`
+        href: `#${key}`
```
Comment (t0nin0s) [inline app/utils/fields-validations.ts]: should be first peer

File: app/controllers/personal-details.ts (hunk 1 above)
Comment (chrisgrimble) [reply app/controllers/personal-details.ts]: Done

File: app/utils/fields-validations.ts (hunk 2 above)
Comment (chrisgrimble) [reply app/utils/fields-validations.ts]: Done

**Pull request merged**